### PR TITLE
feat(ecs): add `availabilityZoneRebalancing` property to EcsService

### PR DESCRIPTION
**Pull request by ren-yamanashi**
### Reason for this change

We can set `availabilityZoneRebalancing` for a Service from Ecs, but this was not supported in the AWS CDK L2 construct.

### Description of changes

Add `availabilityZoneRebalancing` property to BaseServiceOptions and set in the CfnService

### Description of how you validated changes

Added both unit and integration tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts [hunk 1]
```diff
@@ -400,6 +414,14 @@ export interface BaseServiceOptions {
...
    */
   readonly volumeConfigurations?: ServiceManagedVolume[];
+
+  /**
+   * Indicates whether to use Availability Zone rebalancing for the service
+   *
+   * For more information, see [Amazon ECS Service Rebalancing](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-rebalancing.html)
+   * @default - DISABLED
+   */
+  readonly availabilityZoneRebalancing?: AvailabilityZoneRebalancing;
```
Comment (isker) [inline packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts:424]: I don't think it's correct to put it here. You can't use this feature with EXTERNAL services, and ExternalService extends BaseService.

File: packages/aws-cdk-lib/aws-ecs/README.md [hunk 2]
```diff
@@ -1106,6 +1106,36 @@ const service = ecs.FargateService.fromFargateServiceAttributes(this, 'EcsServic
...
+- Services that use the `Replica` strategy
+- Services that specify Availability Zone spread as the first task placement strategy, or do not specify a placement strategy.
+
+Not available for the following:
+
+- Use the `Daemon` strategy
+- Use the `EXTERNAL` boot type (ECS Anywhere)
+- Uses 100% for the `maximumPercent` value.
+- Uses a Classic Load Balancer
+- Uses the `attribute:ecs.availability-zone` as a task placement constraint
```
Comment (isker) [inline packages/aws-cdk-lib/aws-ecs/README.md:1137]: Can't we validate these things and throw at synth time?

**Comment by isker** [comment]
Please link to #32226.

**Comment by isker** [comment]
I asked you those questions because I was partway through implementing this myself when I saw this PR.  I decided to keep going and opened #32263.

**Comment by ren-yamanashi** [comment]
> I asked you those questions because I was partway through implementing this myself when I saw this PR. I decided to keep going and opened #32263.

@isker 

Thank you for your comment.

I apologies, we missed the relevant issue https://github.com/aws/aws-cdk/issues/32226 

After reviewing your comment and https://github.com/aws/aws-cdk/pull/32263, I believe that #32263 is more appropriate and will close this MR.

**Pull request closed without merging**